provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/endpoint/EndpointController.java

```diff
@@ -1,3 +1,165 @@
+    Pair<List<EndpointWrapper>, Integer> root = search(parameters);
+    LinkConfig linkConfig =
+        LinkConfig.builder()
+            .path("Endpoint")
+            .queryParams(parameters)
+            .page(page)
+            .recordsPerPage(count)
+            .totalRecords(root.getRight())
+            .build();
+    return bundler.bundle(
+        BundleContext.of(
+            linkConfig, root.getLeft(), transformer, Endpoint.Entry::new, Endpoint.Bundle::new));
+  }
+
+  /** Read by identifier. */
+  @GetMapping(value = {"/{publicId}"})
+  public Endpoint readByIdentifier(@PathVariable("publicId") String publicId) {
+    return transformer.apply(search(Parameters.forIdentity((publicId))).getKey().get(0));
+  }
+
+  private Pair<List<EndpointWrapper>, Integer> search(MultiValueMap<String, String> parameters) {
+    if (parameters.containsKey("name")) {
+      return searchName(parameters, "name");
+    } else if (parameters.containsKey("identifier")) {
+      return searchName(parameters, "identifier");
+    } else {
+      return searchOrganization(parameters);
+    }
+  }
+
+  /** Search for Identifier. */
+  @GetMapping(params = {"identifier"})
+  public Endpoint.Bundle searchByIdentifier(
+      @RequestParam("identifier") String identifier,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @RequestParam(value = "_count", defaultValue = "15") @Min(0) int count) {
+    return bundle(
+        Parameters.builder()
+            .add("identifier", identifier)
+            .add("page", page)
+            .add("_count", count)
+            .build(),
+        page,
+        count);
+  }
+
+  /** Search by Name. */
+  @GetMapping(params = {"name"})
+  public Endpoint.Bundle searchByName(
+      @RequestParam("name") String name,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @CountParameter @Min(0) int count) {
+    return bundle(
+        Parameters.builder().add("name", name).add("page", page).add("_count", count).build(),
+        page,
+        count);
+  }
+
+  /** Placeholder for Organization search. */
+  @GetMapping(params = {"organization"})
+  public Endpoint.Bundle searchByOrganization(
+      @RequestParam("organization") String organization,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @RequestParam(value = "_count", defaultValue = "15") @Min(0) int count) {
+    throw new UnsupportedOperationException();
+  }
+
+  /** Logic for search using full body from VLER. */
+  private Pair<List<EndpointWrapper>, Integer> searchName(
+      MultiValueMap<String, String> parameters, String function) {
+    String searchFunction = parameters.getFirst(function);
+    AddressResponse addressResponse = vlerClient.endpointByAddress(searchFunction);
+    int filteredCount = 0;
+    List<AddressResponse.Contacts> unfilteredAddressResponsePages =
+        addressResponse.contacts().subList(0, addressResponse.contacts().size());
```

Comment:
It looks like we lost the paging here? Although in practice I doubt there would ever be enough results for us to have multiple pages.

Reply:
Paging added!  Was very broken indeed